configuration/configurator/ConfiguratorMain.cpp

```diff
@@ -0,0 +1,352 @@
+    ::std::cout << "-t -target <target directory>     : directory to which to docs will be written. If not specified, then output will go to ::std::out" << ::std::endl;
+    ::std::cout << "-u -use <schema xsd>              : use specified xsd schema instead of buildset file" << ::std::endl;
+    ::std::cout << "-h -help                          : prints out this usage" << ::std::endl;
+
+    ::std::cout << ::std::endl;
+
+    ::std::cout << ::std::endl << "** Experimental **" << ::std::endl;
+    ::std::cout <<"Example Usage: ./configurator -use esp.xsd -b /opt/HPCCSystems/componentfiles/configxml/ -doc" << ::std::endl;
+    ::std::cout << "-f -file <build set file>         : buildset file name (required if base directory is specfied" << ::std::endl;
+    ::std::cout << "-p -path <base dir path>          : base directory path (required if buildset file name is specified)" << ::std::endl;
+    ::std::cout << "-x -xsd  <xsd file name>          : xsd file name (can be more than one) - For use with buildset file" << ::std::endl;
+    ::std::cout << "-l -list                          : list available xsd files" << ::std::endl;
+    ::std::cout << "-m -xml                           : generate XML configuration file" << ::std::endl;
+    ::std::cout << "-j -json <component key>          : prints JSON" << ::std::endl;
+    ::std::cout << "-c -env -config <path to env xml file> : load environment config xml file (e.g. environment.xml) " << ::std::endl;
+    ::std::cout << "-dump                             : dump out xsd internal structure and values" << ::std::endl;
+}
+
+#ifndef CONFIGURATOR_LIB
+    int main(int argc, char *argv[])
+#else
+    int ConfiguratorMain(int argc, char* argv[])
+#endif // CONFIGURATOR_LIB
+{
+    InitModuleObjects();
+
+    int idx = 1;
+
+    CConfigSchemaHelper *pSchemaHelper = NULL;
+
+    char pBuildSetFile[BUFF_SIZE];
+    char pBuildSetFileDir[BUFF_SIZE];
+    char pTargetDocDir[BUFF_SIZE];
+    char pTargetDocExt[BUFF_SIZE];
+    char pOverrideSchema[BUFF_SIZE];
+    char pBasePath[BUFF_SIZE];
+    char pEnvXMLPath[BUFF_SIZE];
+    StringBuffer strJSONFile;
+    StringBuffer strComponentKey;
+
+    memset(pBuildSetFile, 0, sizeof(pBuildSetFile));
+    memset(pBuildSetFileDir, 0, sizeof(pBuildSetFileDir));
+    memset(pTargetDocDir, 0, sizeof(pTargetDocDir));
+    memset(pOverrideSchema, 0, sizeof(pOverrideSchema));
+    memset(pBasePath, 0, sizeof(pBasePath));
+    memset(pEnvXMLPath, 0, sizeof(pEnvXMLPath));
+
+    strncpy(pTargetDocExt, pDefaultDocExt, sizeof(pTargetDocExt));
+
+    bool bListXSDs      = false;
+    bool bGenDocs       = false;
+	bool bGenJSON       = false;
+    bool bDump          = false;
+    bool bLoadEnvXML    = false;
+
+    StringArray arrXSDs;
+
+    if (argc == 1)
+    {
+        usage();
+        return 0;
+    }
+
+    while(idx < argc)
+    {
+        if (stricmp(argv[idx], "-help") == 0 || stricmp(argv[idx], "-h") == 0)
```

Comment:
I think the standard we're trying to follow is --<full param name> -<abbreviated param name>...
--help -h